pkg/agent/run.go

```diff
@@ -105,21 +114,25 @@ func Run(cmd *cobra.Command, args []string) {
 
 		err := http.ListenAndServe(":8081", server)
 		if err != nil && !errors.Is(err, http.ErrServerClosed) {
-			logs.Log.Fatalf("failed to run the health check server: %s", err)
+			return fmt.Errorf("failed to run the health check server: %s", err)
 		}
-	}()
+		// The agent must stop if the management server stops
+		cancel()
```

Comment:
Instead of calling `cancel`, can we return an error when the server is stopped unexpectedly early?
Normally when the context is not canceled, the server should keep running, and when the context is canceled; we do not have to call `cancel`.

Reply:
Also, when we return an error, the group context will be automatically canceled.

Reply:
I think I agree, that an error should be returned. 
Something like `fmt.Errorf("the API server stopped unexpectedly")`.
But I'll come back to that in another PR.
This `cancel` will work for now.